src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Rectangle implements Figure {
+  shape: string = 'rectangle';
+
+  constructor(
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Invalid width or height');
+    }
+    this.color = color;
+    this.width = width;
+    this.height = height;
```

Comment:
You don't need to set 'color', 'width', and 'height' again, because they are already assigned in the constructor using 'public'. Remove these lines.